src/librustc_passes/error_codes.rs

```diff
@@ -198,6 +198,30 @@ impl Foo for Bar {
 ```
 "##,
 
+E0561: r##"
+A pattern has been used as an argument of a function pointer types.
+
+Erroneous code example:
+
+```compile_fail,E0561
+type A1 = fn(mut arg: u8); // error!
+type A2 = fn(&arg: u32); // error!
+```
+
+When using an alias over a function type, you can't specify things such as an
+argument being mutable, it wouldn't be able to represent functions matching this
+definition.
+
+To fix this issue, just remove patterns (`_` is allowed though). Example:
+
+```
+type A1 = fn(arg: u8); // ok!
+type A2 = fn(_: u32); // ok!
+
+// You can even not provide an argument name:
```

Comment:
I would make this text, not a comment.

Reply:
I was hesitating as well so I'll go as you suggested.